### ISSUE TITLE
Process explorer seems to cause 50-100% CPU spikes on main process

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#137599]
Just opening the process explorer, I see CPU spikes in the main proces (macOS, ARM):
![recording](https://user-images.githubusercontent.com/900690/142753326-d2a23096-5fcb-4154-a0c7-f0c97d231b65.gif)

**Comment by Tyriar** [comment]
I can't repro on my intel mac or windows box, passing back to you if you want to investigate for M1 since I don't have the hardware to action it.

**Comment by deepak1556** [comment]
Can you confirm the version of your `node-gyp` on the machine ? The support for building arm64 on mac came only after [>=7.1.0](https://github.com/nodejs/node-gyp/blob/master/CHANGELOG.md#v710-2020-08-12).

```
> node <path-to-node>/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js --version
```

**Comment by bpasero** [comment]
@deepak1556 yeah I recently installed the latest node.js version:

```
~/.nvm/versions/node/v16.13.0/lib/node_modules/npm/node_modules/node-gyp/bin> node node-gyp.js --version
v8.2.0
```

**Comment by bpasero** [comment]
Actually now I am not so sure anymore we are actually using a native module on macOS for this. 

@Tyriar how do we figure out processes on macOS?

**Comment by deepak1556** [comment]
@bpasero related to your previous comment https://github.com/microsoft/vscode/issues/137599#issuecomment-976664095, as another confirmation can you provide the output of following command

```
> cd vscode
> find ./node_modules -name "*.node" -exec file {} \;
```

**Comment by bpasero** [comment]
@deepak1556 

```
~/Development/Microsoft/monaco> find ./node_modules -name "*.node" -exec file {} \;
./node_modules/vscode-nsfw/build/Release/nsfw.node: Mach-O 64-bit bundle arm64
./node_modules/@vscode/sqlite3/build/Release/sqlite.node: Mach-O 64-bit bundle arm64
./node_modules/mocha/node_modules/fsevents/fsevents.node: Mach-O 64-bit bundle x86_64
./node_modules/node-pty/build/Release/pty.node: Mach-O 64-bit bundle arm64
./node_modules/native-watchdog/build/Release/watchdog.node: Mach-O 64-bit bundle arm64
./node_modules/windows-foreground-love/build/Release/foreground_love.node: Mach-O 64-bit bundle arm64
./node_modules/native-keymap/build/Release/keymapping.node: Mach-O 64-bit bundle arm64
./node_modules/gulp-symdest/node_modules/fsevents/fsevents.node: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit bundle x86_64] [arm64:Mach-O 64-bit bundle arm64]
./node_modules/gulp-symdest/node_modules/fsevents/fsevents.node (for architecture x86_64):	Mach-O 64-bit bundle x86_64
./node_modules/gulp-symdest/node_modules/fsevents/fsevents.node (for architecture arm64):	Mach-O 64-bit bundle arm64
./node_modules/spdlog/build/Release/spdlog.node: Mach-O 64-bit bundle arm64
./node_modules/@parcel/watcher/build/Release/watcher.node: Mach-O 64-bit bundle arm64
./node_modules/@parcel/watcher/prebuilds/darwin-x64/node.napi.glibc.node: Mach-O 64-bit bundle x86_64
./node_modules/@parcel/watcher/prebuilds/win32-x64/node.napi.glibc.node: PE32+ executable (DLL) (GUI) x86-64, for MS Windows
./node_modules/@parcel/watcher/prebuilds/darwin-arm64/node.napi.glibc.node: Mach-O 64-bit bundle arm64
./node_modules/@parcel/watcher/prebuilds/linux-x64/node.napi.musl.node: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, stripped
./node_modules/@parcel/watcher/prebuilds/linux-x64/node.napi.glibc.node: ELF 64-bit LSB shared object, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=2b2144f0fd93b64c28ffbf74a65c96d66fb51ba9, stripped
./node_modules/keytar/build/Release/keytar.node: Mach-O 64-bit bundle arm64
./node_modules/native-is-elevated/build/Release/iselevated.node: Mach-O 64-bit bundle arm64
./node_modules/webpack-stream/node_modules/fsevents/fsevents.node: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit bundle x86_64] [arm64:Mach-O 64-bit bundle arm64]
./node_modules/webpack-stream/node_modules/fsevents/fsevents.node (for architecture x86_64):	Mach-O 64-bit bundle x86_64
./node_modules/webpack-stream/node_modules/fsevents/fsevents.node (for architecture arm64):	Mach-O 64-bit bundle arm64
./node_modules/fsevents/build/Release/.node: Mach-O 64-bit bundle arm64
./node_modules/fsevents/build/Release/fse.node: Mach-O 64-bit bundle arm64
~/Development/Microsoft/monaco>
```

**Comment by deepak1556** [comment]
Thanks they are similar and also everything module is built correctly to the arm architecture.

Now for the original issue, are %CPU readings off between our process explorer and the system activity monitor ?

**Comment by bpasero** [comment]
@deepak1556 nope, `htop` confirms that the main process spikes.

**Comment by deepak1556** [comment]
Tried to repro to my M1 mac, for me the readings from the system monitor and our process explorer are off. I trust the system monitor readings
[Screen Recording 2021-11-24 at 17.02.21.mov.zip](https://github.com/microsoft/vscode/files/7593793/Screen.Recording.2021-11-24.at.17.02.21.mov.zip)

**Comment by bpasero** [comment]
Here you see how htop goes up for me as soon as process explorer starts:
![recording](https://user-images.githubusercontent.com/900690/143199836-719eef69-bd38-47f4-bd05-a60974823386.gif)

Given this is just OSS, I don't mind to close as wontfix. Though it makes development a bit harder since it can result in misleading judgements.

**Comment by Tyriar** [comment]
On macOS we shell out, you're seeing https://github.com/microsoft/vscode/issues/78438#issuecomment-908354643 which is fixed in our product electron build but not the OSS build yet.

**Comment by bpasero** [comment]
@Tyriar maybe, though on macOS ARM I have never seen issues from the libuv issue, but could be that the process explorer is just spawning too many things at once.
